doc/source/advanced.md

```diff
@@ -39,7 +39,45 @@ We recommend the community-maintained [nginx-ingress](https://github.com/helm/ch
 controller, [**kubernetes/ingress-nginx**](https://github.com/kubernetes/ingress-nginx).
 Note that Nginx maintains two additional ingress controllers.
 For most use cases, we recommend the community maintained **kubernetes/ingress-nginx** since that
-is the ingress controller that the development team has the most experience using.
+is the ingress controller that the development team has the most experience using. See below for
+**GCE Ingress** settings.
+
+#### GCE Ingress and manual HTTPS
+To set up HTTPS with GCE Ingress, you must:
+1. offload the proxy, enable `https` (and ideally, change it from 
+ external `LoadBalancer` to close it from outside world - we have ingress for that)
+2. create TLS secrets same as for other Ingresses (e.g. [nginx](https://kubernetes.github.io/ingress-nginx/user-guide/tls/) )
+3. reference the secret in the ingress spec and add `pathSuffix: *`
+
+```yaml
+proxy:
+  https:
+    enabled: true
+    type: offload
+  service:
+    type: NodePort
+ingress:
+  enabled: true
+  annotations:
+    # kubernetes.io/ingress.global-static-ip-name: YOUR-STATIC-IP-NAME
+    kubernetes.io/ingress.class: gce
+    kubernetes.io/ingress.allow-http: "false"
+  hosts:
+  - example.com  # change it to you domain
+  pathSuffix: "*"
+  tls:
+  - hosts:
+    - example.com
+    secretName: jupyterhub-tls
+```
+*Note*: Ingress must have a default backend, it can be equivalent to the default jupyterhub path.
```

Comment:
The GCE Ingress Controller may require this, but I don't require this on my setup with the ingress-nginx-controller.

The key questions about this PR lies here I think.
1. In what way does the GCE ingress controller fail if we lack this default backend?
2. How did you learn about 1? Is there documentation about this?
3. Will adding a default backend influence the function of the ingress-nginx controller?

I'll investigate further, right now, about what I can learn about these things.

Reply:
```suggestion
*NOTE:* Kubernetes ingress resources managed by the GCE Ingress Controller must have a default backend, it can be the same as the default jupyterhub path.
```

Reply:
Regarding the default backend: https://github.com/kubernetes/ingress-gce/blob/master/docs/faq/gce.md#why-do-i-need-a-default-backend

Reply:
This may be useful:

```
gcloud compute http-health-checks list
```

Reply:
https://github.com/kubernetes/ingress-gce/blob/master/docs/faq/gce.md#can-i-configure-gce-health-checks-through-the-ingress

There is a broken link that should point to this deleted documentation, because it was supposedly covered someplace in GCP now, https://github.com/kubernetes/ingress-gce/tree/e906f93959ede5a745ead5c7330a27e5f29a5e2c/examples/health-checks

Reply:
If I read correctly, the ingress controller hardcoding of `/healthz` is for the ingress's *default* backend, which isn't relevant to the Hub's health check endpoint.

Reply:
Hmmm but the default backend will not respond if you have an ingress rule, right?

Hmmm.... This warps my brain.